src/utils/isPlainObj.js

```diff
@@ -5,10 +5,11 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-export default function isPlainObj(value) {
-  return (
-    value &&
-    (typeof value.constructor !== 'function' ||
-      value.constructor.name === 'Object')
-  );
+export default function isPlainObject(value) {
+  if (!value || typeof value !== 'object') {
+    return false;
+  }
+
+  const proto = Object.getPrototypeOf(value);
+  return proto === Object.prototype || proto === null;
```

Comment:
I noticed that [`lodash` is walking up the prototype chain](https://github.com/lodash/lodash/blob/master/isPlainObject.js#L37-L41) when comparing the prototype. Do we have worry about that? Looks like it was added in [this PR](https://github.com/lodash/lodash/pull/3483) although I'm not really sure I understand why it's necessary.

Reply:
Oh because if you have multiple frames you can essentially have `Object.prototype  !== Object.prototype`. That code ensures that an object can still be considered "plain" even if it has a different prototype than your realm's `Object.prototype`. I'd say there's no reason not to incorporate that improvment.